GitUI/GitUIExtensions.cs

```diff
@@ -34,82 +34,90 @@ public static class GitUIExtensions
             return module.GetSingleDiff(firstRevision?.ToString(), secondRevision?.ToString(), file.Name, file.OldName, diffArgs, encoding, true, isTracked);
         }
 
-        [CanBeNull]
-        private static string GetSelectedPatch(
-            [NotNull] this FileViewer diffViewer,
-            [CanBeNull] ObjectId firstRevision,
-            [CanBeNull] ObjectId secondRevision,
-            [NotNull] GitItemStatus file)
+        /// <summary>
+        /// View the changes between the revisions, if possible as a diff
+        /// </summary>
+        /// <param name="fileViewer">Current FileViewer</param>
+        /// <param name="firstId">The first (A) commit</param>
+        /// <param name="selectedRev">The selected (B) commit</param>
+        /// <param name="file">The git item to view</param>
+        /// <param name="defaultText">default text if no diff is possible</param>
+        /// <param name="openWithDiffTool">The difftool command to open with</param>
+        /// <returns>Task to view</returns>
+        public static Task ViewChangesAsync(this FileViewer fileViewer,
```

Comment:
If we make it `this IFileViewer fileViewer`, we should be able to test for of the codepaths

Reply:
There is no interface for the methods FileViewer.cs provides right now (like ViewTextAsync(), ViewGitItemRevisionAsync(), Clear()).
The IFileViewer interface handles FileViewerInternal.cs only right now (but is not used externally.)
This would also require that Module is supplied separetly (or difftool handling is removed as discussed).
So not an obvious change.

Reply:
Ah, I remembered we had `IFileViewer`, it was on a different type...